decisions/01-new-animation-api.md

```diff
@@ -0,0 +1,208 @@
+  }
+  return svgString;
+};
+
+export const settings = {
+  engine: require('@mechanic-design/engine-svg'),
+  mode: 'animation'
+};
+```
+
+```js
+export const handler = async ({ inputs, frame, done }) => {
+  let frameCount = 0;
+  const myDrawLoop = () => {
+    frameCount++;
+    // drawing code
+    if (frameCount >= 100) {
+      done(svgString);
+    } else {
+      frame(svgString);
+      window.requestAnimationFrame(myDrawLoop);
+    }
+  };
+  myDrawLoop();
+};
+
+export const settings = {
+  engine: require('@mechanic-design/engine-svg'),
+  mode: 'customAnimation'
+};
+```
+
+### `engine-react`
+
+```js
+export const handler = async ({ inputs, done, frameCount }) => {
+  if (frameCount >= 100) {
+    done();
+  }
+  return <div></div>;
+};
+
+export const settings = {
+  engine: require('@mechanic-design/engine-react'),
+  mode: 'animation'
+};
+```
+
+The example above re-renders the component for every frame. If a user wants to use something like `react-spring` to speed up animation while keeping the frame-based approach, it could be done like this:
+
+```js
+export const handler = async ({ inputs, done, frameCount }) => {
+  const spring = useSpring({ frameCount });
+
+  if (frameCount >= 100) {
+    done();
+  }
+  return <MyMemoedComponent spring={spring} />;
+};
+
+const MyMemoedComponent = React.memo(({ spring }) => {
+  return <animated.div></animated.div>;
+});
+
+export const settings = {
+  engine: require('@mechanic-design/engine-react'),
+  mode: 'animation'
+};
+```
+
+```js
+export const handler = async ({ inputs, frame, done }) => {
+  const frameCount = myOwnDrawLoopHook();
+  useEffect(() => {
+    if (frameCount < 100) {
+      frame();
+    } else {
+      done();
+    }
+  }, [frameCount]);
+  return <div></div>;
+};
+
+export const settings = {
+  engine: require('@mechanic-design/engine-react'),
+  mode: 'customAnimation'
+};
+```
+
+### `engine-p5`
+
+For `engine-p5`, `mode` can only be set to `static` or `customAnimation`. Or perhaps we default both `animation` and `customAnimation` to the code below?
+
+```js
+export const handler = async ({ inputs, sketch, frame, done }) => {
+  sketch.setup = () => {
+    // do stuff
+  };
+  sketch.draw = () => {
+    // drawing code
+    if (sketch.frameCount < 100) {
+      frame();
+    } else {
+      done();
+    }
+  };
+};
+
+export const settings = {
+  engine: require('@mechanic-design/engine-p5'),
+  mode: 'animation'
+};
+```
+
+## Consequences
+
+### Positive Consequences
+
+The benefits for most users is that they will get an animation API that gets out of their way. The templates and examples for animated design functions will be a lot slimmer, and it will provide us with an API that we can build nice things on top of.
+
+### Negative Consequences
+
+The main disadvantage I can see is:
+
+- Users who are not comfortable writing pure functions are now defaulted into this, and if they opt out, they need to write their own animation code.
```

Comment:
I think this disadvantage can be mitigated by shipping animation loop utils for each engine that a user can just import into their function.

Reply:
Agreed!